### ISSUE TITLE
FULLSKY_SCALAR applies 1/f_sky with no mask, inflating the FG residual (2.4-3.8x measured)

### ISSUE THREAD

**Issue by jrcheshire** [jrcheshire/cmb-augr#51]
### Summary

On the `FULLSKY_SCALAR` spectrum path, `ForecastConfig` applies a scalar `1/f_sky`
correction to every spectrum without requiring that any sky actually be cut. When
`f_sky != 1.0` and no mask or hit map is supplied, the pipeline computes ILC weights
over the full sky (including the Galactic plane), measures the residual over the full
sky, and then *divides* by `f_sky` — a correction that is only meaningful if modes had
been lost to a mask.

`CUTSKY_MC` is self-consistent: `pipeline.py:420` overrides `f_sky` with the realized
`⟨mask⟩`. Only the full-sky path is affected.

### Why this is easy to hit

The most natural entry point produces the inconsistent configuration silently, because
`from_instrument` (`pipeline.py:159`) inherits `f_sky` from the instrument preset, and
every probe-class preset ships `f_sky=0.7`:

```python
from augr.pipeline import ForecastConfig
from augr.config import pico_like
from augr.cleaning import nilc_cleaner

c = ForecastConfig.from_instrument(pico_like(), nilc_cleaner(), nside=128, lmax=192)
print(c.f_sky, c.mask, c.hit_map, c.spectrum_source)
# 0.7 None None SpectrumSource.FULLSKY_SCALAR
```

`pipeline.py:221` then defaults the coverage to uniform full sky:

```python
hit_map = jnp.ones(sky.npix) if config.hit_map is None else config.hit_map
```

while `pipeline.py:237` passes `f_sky=config.f_sky` to `nilc_spectra` unconditionally.

### Impact

Both directions of error land on the foreground residual, and they compound:

1. The ILC weights are derived from an empirical covariance dominated by the Galactic
   plane, where dust is orders of magnitude brighter than at high latitude.
2. The residual spectrum is then measured over that same plane.
3. The `1/f_sky` division inflates the result by a further `1/0.7 = 1.43x`.

Measured in the JPL detector-count study (d1s1, nside=128, 24 bands, NILC and cMILC,
telescope temperatures 30 K and 4 K), the reported foreground residual at the ℓ=88
recombination peak was **2.4–3.8× too high** across every cell of a 20-cell matrix.
Input d1s1 dust BB is 2.8× stronger full-sky than outside a GAL070 cut, and the
residual tracks it. Comparing a residual measured on the full sky against a
systematics threshold defined for the analysis region is a category error.

Notably the weights themselves survive reasonably well — measuring the *existing*
full-sky weights outside the cut recovers nearly the whole effect — so this is
primarily a measurement-region bug rather than a weights-corruption bug. Needlet ILC
weights are spatially local, so plane pixels mostly set plane weights.

### Suggested fix

Refuse the inconsistent combination rather than silently computing it. Something like:
on `FULLSKY_SCALAR` with `mask is None` and `hit_map is None`, raise (or at minimum
warn) unless `f_sky == 1.0`. This mirrors the existing
`Instrument.requires_external_noise` guard, which catches the analogous
"dummy-noise-on-the-analytic-path" footgun by erroring instead of proceeding.

If a scalar-`f_sky` full-sky run is genuinely wanted as a fast approximation, it would
be clearer to have the caller opt in explicitly rather than inherit `f_sky` from the
instrument preset by default.

### Notes

Found while auditing the JPL probe study's detector-count sweep, which reproduced this
pattern in its own driver (`obrient_detector_sweep.py`, fixed JPL-side in
`fabff9e`). The study scripts have moved to a real `|b|` cut with NaMaster MASTER
decoupling and B-purification. Filing here because the library makes the same mistake
reachable by default.
